### PR TITLE
Rust: merge current status of experimental extractor into main

### PR DESCRIPTION
**Pull request by redsun82**
This will require an internal PR for integration of the rust extractor in the internal build system, but should be independent of that for CI purposes.

For ruby, the changes involve getting rid of the fake create workaround as described with more context [here](https://github.com/github/codeql/pull/17434). This involved creating a cargo workspace at the root of `codeql` including the shared tree-sitter extractor and the two extractors using them (that we build with bazel, I'm ignoring `ql` here).

This had the side effect of moving the cargo output directory up in the repository root, which is what causes the other changes around `ruby`.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,9 +5,9 @@ repos:
     rev: v3.2.0
     hooks:
       - id: trailing-whitespace
-        exclude: /test/.*$(?<!\.ql)(?<!\.qll)(?<!\.qlref)|.*\.patch
+        exclude: /test/.*$(?<!\.qlref)|.*\.patch$|.*\.qll?$
       - id: end-of-file-fixer
-        exclude: /test/.*$(?<!\.ql)(?<!\.qll)(?<!\.qlref)|.*\.patch
+        exclude: /test/.*$(?<!\.qlref)|.*\.patch$|.*\.qll?$
```
Comment (geoffw0) [inline .pre-commit-config.yaml:10]: Was `.ql` deleted from this on purpose?

**Review by geoffw0 — COMMENTED** [review]
There seem to be a number of changes in Ruby - we should presumably get a :+1: from someone on the Ruby team to merge this?

**Comment by redsun82** [comment]
> There seem to be a number of changes in Ruby - we should presumably get a 👍 from someone on the Ruby team to merge this?

I'll call it out, good point. It was confirmed by DCA, but it's still good to have eyes on those changes.

**Review by aibaars — APPROVED** [review]
Ruby 👍

**Pull request merged**